src/vehicle.cpp

```diff
@@ -5667,7 +5667,7 @@ bool vehicle::explode_fuel( int p, damage_type type )
             pgettext("memorial_female", "The fuel tank of the %s exploded!"),
             name.c_str());
         g->explosion( global_part_pos3( p ), pow, 0.7, data.fiery_explosion );
-        parts[p].hp = 0;
+        mod_hp( parts[p], 0 - parts[ p ].hp(), DT_HEAT );
```

Comment:
Possibly clearer although I do have another call using the unary operator so that's inconsistent.